public/react/Configure/components/Teams/services/team-configuration/hooks/sanitizeAllTeamData.ts

```diff
@@ -0,0 +1,50 @@
+import { type Configuration, createUpdatedTeamData } from "../team-configuration";
+
+export function sanitizeAllTeamData(
+  allTeamData: any,
+  teamName: string,
+  hierarchyLevel: string,
+  updates: Configuration
+) {
+  const allUpdates = createUpdatedTeamData(allTeamData, {
+    teamName,
+    hierarchyLevel,
+    configuration: updates,
+  });
+
+  const sanitized = entriesFlatMap(allUpdates, ([teamKey, teamConfig]) => {
+    if (!teamConfig) return [];
+
+    const configs = entriesFlatMap(teamConfig, ([configKey, config]) => {
+      if (!config) return [];
+
+      const sanitizedConfig = filterNullValues(config);
+      return notEmpty(sanitizedConfig, [configKey, sanitizedConfig]);
+    });
+
+    return notEmpty(configs, [teamKey, configs]);
+  });
+
+  return sanitized;
+}
+
+function entriesFlatMap<T, R extends readonly [string, any]>(
```

Comment:
Can we name `T` and `R` something semantic

same below

Reply:
these funcs were just supposed to make the sanitize func cleaner and easier to read, figuring out what the types need to be for them is more of a hassle than I wanna fool with, I just inlined them in the sanitize func